File: DESC_Python/src/specprocess_readpara.py
def specprocess_readpara(filename):
    """
    Read parameter file and return parameter dictionary P.

    The file format mirrors the MATLAB version; lines starting with '*' are comments.
    Parameters are expected in order, one per non-comment line.
    """
    P = {}
    with open(filename, 'r') as fid:
        npara = 0
        for line in fid:
            line = line.strip('\n')
            if not line or line.startswith('*'):
                continue
            npara += 1
            if npara == 1:  # event directory
                P['evdir'] = line
            elif npara == 2:  # target phase
                P['target'] = line
            elif npara == 3:  # maximum distance
                x = list(map(float, line.split()))
                P['delmax'] = x[0]
            elif npara == 4:  # magnitude range
                x = list(map(float, line.split()))
                P['minmag'], P['maxmag'] = x[0], x[1]
            elif npara == 5:  # three frequency bands
                x = list(map(float, line.split()))
                P['freq1'] = [x[0], x[1]]
                P['freq2'] = [x[2], x[3]]
                P['freq3'] = [x[4], x[5]]
            elif npara == 6:  # STN for the three bins
                x = list(map(float, line.split()))
                P['stnmin'] = [float(__import__('math').log10(v)) for v in x[:3]]
            elif npara == 7:  # fraction of events to check
                x = list(map(float, line.split()))
                P['randfrac'] = x[0]
            elif npara == 8:  # area limits
                x = list(map(float, line.split()))
                P['qlat1'], P['qlat2'], P['qlon1'], P['qlon2'] = x[0], x[1], x[2], x[3]
            elif npara == 9:  # error type
                x = list(map(float, line.split()))
                P['itype'] = int(x[0])
            elif npara == 10:  # number of iterations
                x = list(map(float, line.split()))
                P['nitmax'] = int(x[0])
            elif npara == 11:  # time interval for travel-time bin stack
                x = list(map(float, line.split()))
                P['dtt'] = x[0]
            elif npara == 12:  # tstar frequency range
                x = list(map(float, line.split()))
                P['fmin_tstar'], P['fmax_tstar'] = x[0], x[1]
            elif npara == 13:  # minimum average stn for tstar
                x = list(map(float, line.split()))
                import math
                P['minavgstn'] = math.log10(x[0])
            elif npara == 14:  # target network(s)
                P['targnet'] = line.split()
            elif npara == 15:  # target station(s)
                P['targsta'] = line.split()
            elif npara == 16:  # target channel(s)
                P['targchan'] = line.split()
            elif npara == 17:  # eqinfo path
                P['eqinfo'] = line
    # Default eqinfo if not provided explicitly
    if 'eqinfo' not in P and 'evdir' in P:
        P['eqinfo'] = f"{P['evdir']}/eqinfo.mat"
    return P

File: DESC_MATLAB/src/specprocess_readpara.m
function P = specprocess_readpara(filename)

fid = fopen(filename,'r');

npara = 0;
while (1 == 1)
    a = fgetl(fid);
    if (a == -1) break; end
    if (a(1:1) ~= '*')
        npara = npara+1;
        if (npara == 1)    % this is for event directory
            P.evdir = a;
        elseif (npara==2)  % this is for target Phase.
            P.target = a;
        elseif (npara==3)
            x = sscanf(a,'%f'); % this is for maximum distance
            P.delmax = x(1);
        elseif (npara == 4)
            x = sscanf(a,'%f'); % this is for magnitude range
            P.minmag = x(1);
            P.maxmag = x(2);
        elseif (npara == 5)
            x = sscanf(a,'%f'); % this is for the three frequency band
            P.freq1(1) = x(1);
            P.freq1(2) = x(2);
            P.freq2(1) = x(3);
            P.freq2(2) = x(4);
            P.freq3(1) = x(5);
            P.freq3(2) = x(6);
        elseif (npara == 6)
            x = sscanf(a,'%f'); % this is for STN for the three bins
            P.stnmin = log10(x(1:3));
        elseif (npara == 7)
            x = sscanf(a,'%f'); % this is for fraction of events to check, 1 for all
            P.randfrac = x(1);
        elseif (npara == 8)
            x = sscanf(a,'%f'); % this is for limits of area
            P.qlat1 = x(1);
            P.qlat2 = x(2);
            P.qlon1 = x(3);
            P.qlon2 = x(4);
        elseif (npara == 9)
            x = sscanf(a,'%f'); % this is for error type.
            P.itype = x(1);
        elseif (npara == 10)
            x = sscanf(a,'%f'); % this is for number of iterations
            P.nitmax = x(1);
        elseif (npara == 11)
            x = sscanf(a,'%f');  % this is for time interval for travel-time bin stack
            P.dtt = x(1);
        elseif (npara == 12)
            x = sscanf(a,'%f'); % this is for tstar frequency range
            P.fmin_tstar = x(1);
            P.fmax_tstar = x(2);
        elseif (npara == 13)
            x = sscanf(a,'%f'); % this is for minium average stn for tstar
            P.minavgstn = log10(x(1));
        elseif (npara == 14)  % * means all networks
            x = strsplit(a,' ');
            P.targnet = x;
        elseif (npara == 15)  % * means all stations
            x = strsplit(a,' ');
            P.targsta = x;
        elseif (npara == 16)
            x = strsplit(a,' ');
            P.targchan = x;
        elseif (npara == 17)
            P.eqinfo = a;
        end
    end
end

if (npara == 15)
    P.eqinfo = [P.evdir,'/eqinfo.mat'];
end

